packages/flutter_tools/lib/src/runner/local_engine.dart

```diff
@@ -50,6 +50,10 @@ class LocalEngineLocator {
     String? localWebSdk,
     String? packagePath,
   }) async {
+    if (localHostEngine != null && localEngine == null) {
+      throwToolExit(_userMessages.runnerHostEngineRequiresLocalEngine, exitCode: 2);
```

Comment:
Why exit code 2? Is there a constant for this somewhere if it means something special? At the very least, let's define a local constant with a meaningful name.

Reply:
This was done for consistency - all other calls to `throwToolExit` in this file use exit code 2.

Most other `throwToolExit` calls in flutter_tools also use exit code 2 (although there are a few places that use other exit codes).

I don't know the history of why the tool uses that number, and there is no constant defined for this.  It would be helpful to make this more consistent throughout the tool, but that's out of scope for this PR.